src/dynamodb.js

```diff
@@ -52,8 +71,34 @@ async function destroyTables (dynamoClient, uniqueIdentifier) {
     tablesToDestroy
       .map(async TableName => {
         try {
-          await dynamoClient.deleteTable({ TableName }).promise();
-          await dynamoClient.waitFor('tableNotExists', { TableName }).promise();
+          const deleteRequest = dynamoClient.deleteTable({ TableName });
+
+          deleteRequest.on('retry', response => {
+            // Do not retry table deletion because it leads to
+            // ResourceNotFoundException when trying to delete the table twice
+            response.error.retryable = false;
+          });
+
+          try {
+            await deleteRequest.promise();
+          } catch (e) {
+            if (e.code === 'TimeoutError') {
+              await waitForReady(`Deleted table ${TableName}`, async () => {
+                try {
+                  await dynamoClient.describeTable({ TableName }).promise();
+                  throw new Error(`Table ${TableName} still exists`);
```

Comment:
Instead of using exceptions for flow control here, what about handling the wait with a `while(true)` loop?

Reply:
waitForReady has the logic that retries after a timeout. It might be nice to have a `waitForException` helper which does the opposite.